linchpin/examples/quick-start-with-controller/README.md

```diff
@@ -0,0 +1,357 @@
+
+```
+--public-key-upload <PATH_TO_THE_PUBLIC_KEY> --key-name mykey
+```
+
+If the key is not the default one used for ssh connections you may need to set it for ansible access to the provisioned hosts (configured via target inventory - `linchpin/inventories/quick-start.inventory` in our example) by adding the path to the private key, for example:
+```
+--public-key-upload <PATH_TO_THE_PUBLIC_KEY> --key-name mykey --ansible-private-key <PATH_TO_THE_PRIVATE_KEY>
+```
+
+In this case you probably don't want to use the private key as [master's ssh key](../../README.md#ssh-keys). If you do, add also `--key-use-for-master` option.
+
+##### Watch the running tests with virt-manager.
+
+Add public key for virt-manager connection to the runners' authorized keys by copying them into a special directory of [kstest runner role](../../../ansible/roles/kstest) and running the kstest runners deployment playbook:
+
+```
+cp <PATH_TO_THE_PUBLIC_KEY> ansible/roles/kstest/files/authorized_keys
+ansible-playbook -i linchpin/inventories/quick-start.inventory ansible/kstest-runners-deploy.yml --tags "ssh-keys"
+```
+Only the playbook tasks tagged "ssh-keys" are run to be faster.
+
+To see the IP addresses of runners for adding connections to virt-manager run:
+```
+./kstests-in-cloud.sh status quick-start --show-inventory
+```
+
+##### Use different cloud profile.
+
+The cloud profile is used for provisioning and destoying of runners (or even controller) in cloud.
+
+Default location of cloud configuration file for controller deployment is:
+```
+~/.config/openstack/clouds.yml
+```
+from where the configuration is passed to the controller to default location for linchpin which used to provision master and runners:
+```
+~/.config/linchpin/clouds.yml
+```
+
+Default profile is `kstests`. To use another profile set `--cloud` option of `kststs-in-cloud.sh` (eg in [test.sh](test.sh)).
+
+
+### Troubleshooting
+
+See also [Status](#status)
+
+##### Check openstack cloud access.
+
+Having `~/.config/openstack/clouds.yml` configured with `kstests` profile, to check the access run:
+
+```
+openstack --os-cloud kstests server list
+```
+
+##### Stop running test.
+
+If the test is run from controller as in [run.sh](run.sh) or [test.sh](test.sh) just interrupt (CTRL-C) or terminate the ansible process.
+
+If the test was scheduled from controller stop the service:
+```
+systemctl --user stop kstests-cloud-quick-start.service
+```
+
+If the test was scheduled from master just [destroy](destroy.sh) the runners.
+
+##### Clean up interrupted test or provisioning.
+
+For now the only really safe option after [stopping a running test](#stop-running-test) is [destroying](destroy.sh) the target and re-provisioning the target.
+
+I'd like to add playbooks for better interrupted or brokenm test handling:
+
+- stop the test run with processing of results obtained so far
```

Comment:
I guess in the future we could fetch the results to master incrementally, right after a given kickstart test finishes running ?

this has quite a few benefits:
- if you are watching a test run, you can investigate failures on master once their logs are fetched, you don't have to look for them on the runners (which might not be easy to manualy SSH to, especially if dynamically provisioned) or wait for the full run to finish
- if the runners crash and burn, you still have all the logs up till that point
- the full run should be faster, as we are making optimal use of the time to both run tests and download results, instead of all result downloads happening at the end in series (IIRC) over all runners

Reply:
This would have to be changed (support added) in run_kickstart_tests.sh.

Reply:
Yep, just brainstorming a bit. :)